src/datanode/src/datanode.rs

```diff
@@ -86,7 +87,10 @@ impl Datanode {
         if let Some(task) = &self.heartbeat_task {
             // Safety: The event_receiver must exist.
             let receiver = self.region_event_receiver.take().unwrap();
-            task.start(receiver).await?;
+            let notify = Arc::new(Notify::new());
```

Comment:
Why need to wait for the first response?  If the datanode needs some information at startup, we should issue an RPC synchronously.

What would happen if the first heartbeat response was delayed or not properly processed?

Reply:
I'm still thinking about this problem. I added this feature because I am trying to avoid the outside seeing some intermeddle states. It needs the first heartbeat's response to close read-only regions or set a `writable` flag for regions.  How about adding a timeout for the `notify.notified`. ?